protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java

```diff
@@ -78,6 +78,18 @@ public static void setExtensionRegistry(ExtensionRegistryLite newRegistry) {
     globalRegistry = checkNotNull(newRegistry, "newRegistry");
   }
 
+  /**
+   * Local cache of buffers to use for parsing.  ThreadLocal used a WeakReference internally, so
```

Comment:
I originally thought that it needed to use a WeakRef, (and it isn't clear from TL). That the value will be cleaned up if under pressure. even if the thread exists from my understanding.